client/src/components/signup/SignupForm.js

```diff
@@ -0,0 +1,128 @@
+import { useState } from 'react'
+import SignupFormButton from '../button/SignupFormButton'
+import SignupFormField from './SignupFormField'
+import userIcon from '../../icons/signup/signup-user-icon.png'
+import emailIcon from '../../icons/signup/signup-email-icon.png'
+import passwordIcon from '../../icons/signup/signup-password-icon.png'
+import styles from '../../styles/Signup-styles/SignupForm.module.css'
+
+function SignupForm({ formChoice, onValidatedSubmit }) {
+  // Create state variable to store:
+  //    -current credentials from inputs in string (name, email, password)
+  const [inputValues, setInputValues] = useState({
+    username: '',
+    email: '',
+    password: '',
+  })
+
+  //    -current error message for each field in a form of a object of strings
+  const [errors, setErrors] = useState({
+    username: '',
+    email: '',
+    password: '',
+  })
+
+  // isFormValid function is a function that runs only in case of the "signup" option.
+  // This functions checks all the fields and updates the error messages accordingly.
+  // This function returns true if all the fields are vaid and false when at least one
+  // field is invalid.
+  const isFormValid = () => {
+    // Create temporary variables to store the validation status of each field
+    const fieldValidationErrors = {}
+    const isValidUsername = inputValues.username.length >= 5
+    const isValidEmail = (/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i).test(inputValues.email)
+    const isValidPassword = inputValues.password.length >= 5
+
+    // Based on the above validation assign the error message, or remove the error message
+    // in the fieldValidationErrors variable.
+    fieldValidationErrors.username = isValidUsername ? '' : 'minimum 5 characters'
+    fieldValidationErrors.email = isValidEmail ? '' : 'invalid email address'
+    fieldValidationErrors.password = isValidPassword ? '' : 'minimum 5 characters'
+
+    // Assign the saved errors (fieldValidationErrors) to the errors state variable.
+    setErrors({ ...fieldValidationErrors })
```

Comment:
How about we set the errors only when ` if (isValidUsername && isValidEmail && isValidPassword) { return true }` is false. We can set the error state here before returning false because that is when it will matter, right?

Reply:
Yes that makes sense.
The only thing will be that when all the credentials are correct the error messages will not update/clear.
However that won't be important anymore as we will be redirected to the chats page

Reply:
Ok, that makes sense.